File: src/main/java/api/carpooling/application/service/AuthService.java
package api.carpooling.application.service;

import api.carpooling.application.dto.auth.LoginUserRequest;
import api.carpooling.application.dto.auth.RegisterUserRequest;
import api.carpooling.application.dto.user.UserDTO;

/**
 * Service interface for user authentication operations.
 */
public interface AuthService {

    /**
     * Registers a new user.
     *
     * @param registerUserRequest registration request payload
     * @return UserDTO of the registered user
     */
    UserDTO register(RegisterUserRequest registerUserRequest);

    /**
     * Authenticates a user with email and password.
     *
     * @param loginUserRequest login request payload
     * @return UserDTO of the authenticated user
     */
    UserDTO login(LoginUserRequest loginUserRequest);

    /**
     * Refreshes JWT access token using a refresh token.
     *
     * @param refreshToken the refresh token
     * @return UserDTO with new JWT token
     */
    UserDTO refreshToken(String refreshToken);

    /**
     * Retrieves the profile of the authenticated user.
     *
     * @param token JWT token from Authorization header
     * @return UserDTO of the authenticated user
     */
    UserDTO me(String token);
}

File: frontend/carpooling-frontend/src/services/authService.js
import api from "./api";

export const login = async (email, password) => {
    const response = await api.post("/auth/login", { email, password });

    const user = response.data;

    localStorage.setItem("token", user.token);
    localStorage.setItem("refreshToken", user.refreshToken);

    return user;
};

export const register = async (username, email, password, phoneNumber) => {
    const response = await api.post("/auth/register", {
        username,
        email,
        password,
        phoneNumber
    });

    const user = response.data;

    localStorage.setItem("token", user.token);
    localStorage.setItem("refreshToken", user.refreshToken);

    return user;
};

export const getCurrentUser = async () => {
    const token = localStorage.getItem("token");

    if (!token) return null;

    const response = await api.get("/auth/me", {
        headers: { Authorization: `Bearer ${token}` }
    });

    return response.data;
};

export const logout = async () => {
    const token = localStorage.getItem("token");
    if (token) {
        await api.post("/auth/logout", null, {
            headers: { Authorization: `Bearer ${token}` }
        });
    }
    localStorage.clear();
};
